core/sdk-core/src/main/java/software/amazon/awssdk/core/progress/listener/ProgressListener.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.progress.listener;
+
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPreviewApi;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.core.progress.snapshot.ProgressSnapshot;
+import software.amazon.awssdk.http.SdkHttpRequest;
+
+/**
+ * The {@link ProgressListener} interface may be implemented by your application in order to receive event-driven updates on
+ * the progress of a service call. When you construct an {@link PutObjectRequest} or {@link
+ * UploadPartRequest} request submitted to the Sdk, you may provide a variable number of {@link
```

Comment:
Maybe we should say "An {@link SdkRequest} like a {@link PutObjectRequest}..."

Reply:
I thought of having that, and went with SdkHttpRequest as the "requestPrepared" method deals with requests that have been marshalled into as SdkHttpRequest already.

 I do see your point that technically, the SdkRequest is what was submitted. I can change this